### ISSUE TITLE
Broken when using with latest AndroidX material alpha05 and appcompat alpha04

### ISSUE THREAD

**Issue by yccheok** [Gericop/Android-Support-Preference-V7-Fix#198]
All the time, this wonderful lib works well with the following combination.

```
implementation 'com.takisoft.fix:preference-v7:28.0.0.0'
implementation 'androidx.appcompat:appcompat:1.1.0-alpha03'
implementation 'com.google.android.material:material:1.1.0-alpha04'

<style name="Theme.WeNotePreference.White" parent="@style/PreferenceFixTheme.Light.NoActionBar">

```

However, when I upgrade my AndroidX version to

```
implementation 'com.takisoft.fix:preference-v7:28.0.0.0'
implementation 'androidx.appcompat:appcompat:1.1.0-alpha04'
implementation 'com.google.android.material:material:1.1.0-alpha05'

<style name="Theme.WeNotePreference.White" parent="@style/PreferenceFixTheme.Light.NoActionBar">
```

Thing broken as below. Note, for the following screenshot, it already scrolled to the end of list. As you can see, the last item is being covered up.

![0](https://user-images.githubusercontent.com/308276/55865088-49ada800-5bb0-11e9-96e7-6af7ca42f3fd.png)

My WILD guess is, maybe I need to remove `implementation 'com.takisoft.fix:preference-v7:28.0.0.0'` and use

`implementation 'com.takisoft.preferencex:preferencex:1.0.0'`

and replace `com.takisoft.fix.support.v7.preference.PreferenceFragmentCompat` with `com.takisoft.preferencex.PreferenceFragmentCompat`

But, I have no idea, what theme I should use, as `@style/PreferenceFixTheme.Light.NoActionBar` is no longer available.

Thanks.

**Comment by gregkorossy** [comment]
`PreferenceFixTheme` is no longer needed (nor available). Just use an `AppCompat` / `MaterialComponents` theme.

**Comment by yccheok** [comment]
Ok. Thanks. I'm now using `@style/Theme.AppCompat.Light.NoActionBar`

But, even I use `implementation 'com.takisoft.preferencex:preferencex:1.0.0'`, it still yield same problem (Unable to scroll till the end, in API 28 device)

Seem like issue in AndroidX preference?

This is the end part of my preference.xml

```
    <PreferenceCategory
        android:title="@string/preference_about">
        <Preference
            android:key="RATE_APP"
            android:title="@string/preference_rate_app_title"
            android:summary="@string/preference_rate_app_summary"
            app:singleLineTitle="false" />
        <Preference
            android:key="SHARE_APP"
            android:title="@string/preference_share_app_title"
            app:singleLineTitle="false" />
        <Preference
            android:key="TRANSLATIONS"
            android:title="@string/preference_translations_title"
            android:summary="@string/preference_translations_summary"
            app:singleLineTitle="false" />
        <Preference
            android:key="FACEBOOK_PAGE"
            android:title="@string/preference_facebook_page_title" />
        <Preference
            android:key="TRELLO_BOARD"
            android:title="@string/preference_trello_board_title"
            app:singleLineTitle="false" />
        <Preference
            android:key="PRIVACY_POLICY"
            android:title="@string/preference_privacy_policy_title"
            app:singleLineTitle="false" />
        <Preference
            android:key="VERSION"
            android:title="@string/preference_version"
            app:singleLineTitle="false" />
    </PreferenceCategory>
</PreferenceScreen>
```

**Comment by gregkorossy** [comment]
What's your layout XML? I think the preference fragment's height is set to match_parent while the toolbar is also there, so the total size of your layout is match_parent + toolbar height.

**Comment by yccheok** [comment]
This is how my XML looks like. It works fine all the time, unless I upgrade to "AndroidX material alpha05 and appcompat alpha04"

```
<?xml version="1.0" encoding="utf-8"?>
<androidx.coordinatorlayout.widget.CoordinatorLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    android:layout_width="match_parent"
    android:layout_height="match_parent">

    <com.google.android.material.appbar.AppBarLayout
        android:id="@+id/app_bar_layout"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        android:theme="?attr/appBarLayoutTheme">

        <FrameLayout
            android:layout_width="match_parent"
            android:layout_height="wrap_content">
            <androidx.appcompat.widget.Toolbar
                android:id="@+id/toolbar"
                android:layout_width="match_parent"
                android:layout_height="?attr/actionBarSize"
                android:background="?attr/colorPrimary"
                app:popupTheme="@style/AppTheme.PopupOverlay" />

            <include layout="@layout/indeterminate_smooth_progress_bar"/>
        </FrameLayout>
    </com.google.android.material.appbar.AppBarLayout>

    <FrameLayout
        android:id="@+id/content"
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        app:layout_behavior="@string/appbar_scrolling_view_behavior" />

</androidx.coordinatorlayout.widget.CoordinatorLayout>
```

This is the code being called in fragment activity's `onCreate`

```
        if (savedInstanceState == null) {
            // Display the fragment as the main content.
            PreferenceFragment preferenceFragment = PreferenceFragment.newInstance();
            this.getSupportFragmentManager().beginTransaction().replace(R.id.content, preferenceFragment).commit();
        }
```

**Comment by yccheok** [comment]
I use the following XML and the problem is gone. Thanks.

```
<?xml version="1.0" encoding="utf-8"?>
<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    android:orientation="vertical"
    android:layout_width="match_parent"
    android:layout_height="match_parent">

    <com.google.android.material.appbar.AppBarLayout
        android:id="@+id/app_bar_layout"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        android:theme="?attr/appBarLayoutTheme">

        <FrameLayout
            android:layout_width="match_parent"
            android:layout_height="wrap_content">
            <androidx.appcompat.widget.Toolbar
                android:id="@+id/toolbar"
                android:layout_width="match_parent"
                android:layout_height="?attr/actionBarSize"
                android:background="?attr/colorPrimary"
                app:popupTheme="@style/AppTheme.PopupOverlay" />

            <include layout="@layout/indeterminate_smooth_progress_bar"/>
        </FrameLayout>
    </com.google.android.material.appbar.AppBarLayout>

    <FrameLayout
        android:id="@+id/content"
        android:layout_width="match_parent"
        android:layout_height="match_parent" />

</LinearLayout>
```

**Comment by gregkorossy** [comment]
I've just tried the sample with CoordinatorLayout and AppBarLayout and it works for me. Don't know what was the problem with yours.

Also, some teaser...
![Screenshot_1555335893](https://user-images.githubusercontent.com/5181621/56137739-c1efe100-5f95-11e9-87c4-03678e281006.png) ![Screenshot_1555335896](https://user-images.githubusercontent.com/5181621/56137749-c4ead180-5f95-11e9-8917-d770a8186703.png)

**Comment by gregkorossy** [comment]
1.1.0-alpha04 is out. The README is to-be-updated.

**Comment by gregkorossy** [comment]
Also, 1.1.0-alpha05 is out as a quick fix for appcompat 1.1.0-alpha05 (contains fragment 1.1.0-alpha08 which is the actual cause) being incompatible with preferences 1.1.0-alpha01 through 1.1.0-alpha04.
